fem/dgmassinv_kernels.hpp

```diff
@@ -172,6 +179,43 @@ real_t DGMassDot(const int e,
    return s_dot[0];
 }
 
+template<int T_D1D = 0>
+MFEM_HOST_DEVICE inline
+void DGMassBasis1D(const int e,
+                   const int NE,
+                   const real_t *b_,
+                   const real_t *x_,
+                   real_t *y_,
+                   const int d1d = 0)
+{
+   const int D1D = T_D1D ? T_D1D : d1d;
+
+   const auto b = Reshape(b_, D1D, D1D);
+   const auto x = Reshape(x_, D1D, NE);
+   auto y = Reshape(y_, D1D, NE);
+
+   constexpr int MD1 = T_D1D ? T_D1D : DofQuadLimits::MAX_D1D;
+   real_t Y[MD1];
+
+   MFEM_FOREACH_THREAD(i,x,D1D)
+   {
+      real_t val = 0.0;
+      MFEM_FOREACH_THREAD(j,y,D1D)
+      {
+         val += b(i,j)*x(j,e);
+      }
+      Y[i] = val;
```

Comment:
I feel this may be correct by "chance", I think it would be more appropriate to use a `for` loop, otherwise the expectation is that `threadIdx.y == 1`, but that makes the code unnecessarily difficult to understand if I'm not mistaken.
```suggestion
      real_t val = 0.0;
      for (int j = 0; j < D1D; ++j)
      {
         val += b(i,j)*x(j,e);
      }
      Y[i] = val;
```

Reply:
Good catch, fixed now.